mailet/api/src/test/java/org/apache/mailet/AttributeValueTest.java

```diff
@@ -430,9 +474,19 @@ void fromJsonStringShouldThrowOnMalformedFormattedJson() {
     @Nested
     class ListSerialization {
         @Test
-        void nullStringListShouldThrowAnException() {
-            assertThatNullPointerException()
-                .isThrownBy(() -> AttributeValue.ofAny((List<String>) null));
+        void nullListShouldThrowAnException() {
+            assertThatNullPointerException().
+                isThrownBy(() -> AttributeValue.of((List<AttributeValue<?>>) null));
+        }
+
+        @Test
+        void stringListShouldThrowAnException() {
+            assertThatThrownBy(() -> AttributeValue.ofAny((List<String>) ImmutableList.of("not an AttributeValue")));
```

Comment:
could you add `isInstanceOf()` and `hasMessage()` assertion? According this test case, it should throw `ClassCastException`, but not the Exception from Precondition.check() of your 
```java
public static AttributeValue<Collection<AttributeValue<?>>> of(Collection<AttributeValue<?>> value) {}
```

idem for some elsewhere

Reply:
Because it really does :/ I did not find a way to force the stream to do not cast it before `.allMatch`